src/status_im/chat/models/delete_message.cljs

```diff
@@ -0,0 +1,90 @@
+(ns status-im.chat.models.delete-message
+  (:require [re-frame.core :as re-frame]
+            [status-im.chat.models.message-list :as message-list]
+            [status-im.ethereum.json-rpc :as json-rpc]
+            [status-im.utils.datetime :as datetime]
+            [status-im.utils.fx :as fx]
+            [taoensso.timbre :as log]))
+
+(defn- update-db-clear-undo-timer
+  [db chat-id message-id]
+  (when (get-in db [:messages chat-id message-id])
+    (update-in db
+               [:messages chat-id message-id]
+               dissoc
+               :deleted-undoable-till)))
+
+(defn- update-db-delete-locally
+  "Delete message in re-frame db and set the undo timelimit"
+  [db chat-id message-id undo-time-limit-ms]
+  (when (get-in db [:messages chat-id message-id])
+    (update-in db
+               [:messages chat-id message-id]
+               assoc
+               :deleted? true
+               :deleted-undoable-till (+ (datetime/timestamp)
+                                         undo-time-limit-ms))))
+
+(defn- update-db-undo-locally
+  "Restore deleted message if called within timelimit"
+  [db chat-id message-id]
+  (let [{:keys [deleted? deleted-undoable-till]}
+        (get-in db [:messages chat-id message-id])]
+    (if (and deleted?
+             (> deleted-undoable-till (datetime/timestamp)))
+      (update-in db
+                 [:messages chat-id message-id]
+                 dissoc
+                 :deleted?
+                 :deleted-undoable-till)
+      (update-db-clear-undo-timer db chat-id message-id))))
+
+(fx/defn delete
+  "Delete message now locally and broadcast after undo time limit timeout"
+  {:events [:chat.ui/delete-message]}
+  [{:keys [db]} {:keys [chat-id message-id]} undo-time-limit-ms]
+  (when (get-in db [:messages chat-id message-id])
+    (assoc
+     (message-list/rebuild-message-list
+      {:db (update-db-delete-locally db chat-id message-id undo-time-limit-ms)}
+      chat-id)
+     :utils/dispatch-later [{:dispatch [:chat.ui/delete-message-and-send
+                                        {:chat-id    chat-id
+                                         :message-id message-id}]
+                             :ms       undo-time-limit-ms}])))
+
+(fx/defn undo
+  {:events [:chat.ui/undo-delete-message]}
+  [{:keys [db]} {:keys [chat-id message-id]}]
+  (when (get-in db [:messages chat-id message-id])
+    (message-list/rebuild-message-list
+     {:db (update-db-undo-locally db chat-id message-id)}
+     chat-id)))
+
+(fx/defn delete-and-send
+  {:events [:chat.ui/delete-message-and-send]}
+  [{:keys [db]} {:keys [message-id chat-id]}]
+  (when (get-in db [:messages chat-id message-id])
+    {:db             (update-db-clear-undo-timer db chat-id message-id)
+     ::json-rpc/call [{:method      "wakuext_deleteMessageAndSend"
+                       :params      [message-id]
+                       :js-response true
+                       :on-error    #(log/error "failed to delete message " %)
```

Comment:
> it could be a good idea to log the message ID
> can we trust the backend is logging all the necessary data in case an RPC call fails? What information is contained in the RPC error object? 

There's definitely no message ID in RPC error message if passing a wrong type of message ID (JSON unmarshal error), so better add message ID here

Reply:
I see from recent commits you added the message ID to the log in case an error happens, so we're gold now ;)

----

The following is not a suggestion for change, but rather something you might like to use, just as I do. I usually prefer to log errors by passing a human readable string, but also an additional map to contextualize the error in a parseable way. The timbre lib allows (and even encourages) us to this. Of course there are tons of nuances for how a team decides what to log, but you get the idea. For example:

```clojure
#(log/error "Failed to delete message" {:message-id message-id
                                        :error      %})
```